### PR TITLE
add ngc signing job for auto signing

### PR DESCRIPTION
**Pull request by shivakunv**
add ngc signing job for auto signing
@cdesiniotis @elezar

successfull pipelines: https://gitlab-master.nvidia.com/dl/container-dev/k8s-driver-manager/-/merge_requests/3

### CODE REVIEW

File: .nvidia-ci.yml [hunk 1]
```diff
@@ -148,3 +148,55 @@ release:ngc-ubi8:
...
+  script:
+    - 'echo "Signing the image ${IMAGE_NAME}:${IMAGE_TAG}"'
+    - ngc-cli/ngc registry image publish --source ${IMAGE_NAME}:${IMAGE_TAG} ${IMAGE_NAME}:${IMAGE_TAG} --public --discoverable --allow-guest --sign --org nvidia 
+
+sign:ngc:
+  extends:
+    - .sign:ngc
+  parallel:
+    matrix:
+    - SIGN_JOB_NAME: ["" ]
```
Comment (elezar) [inline .nvidia-ci.yml:194]: nit:
```suggestion
    - SIGN_JOB_NAME: [""]
```

File: .nvidia-ci.yml [hunk 2]
```diff
@@ -148,3 +148,55 @@ release:ngc-ubi8:
...
+      DIST: ["CI_COMMIT_TAG", "ubi8"]
+  rules:
+    - if: '$DIST == "CI_COMMIT_TAG"'
```
Comment (elezar) [inline .nvidia-ci.yml:197]: One thought on this. Does something like `COMMIT_TAG_ONLY` make sense here for better readability.

Not a blocker though.

**Review by elezar — APPROVED** [review]
I have a minor comment / question.

Looks good otherwise.

File: .nvidia-ci.yml (hunk 2 above)
Comment (shivakunv) [reply .nvidia-ci.yml:197]: not needed . 
removed

File: .nvidia-ci.yml [hunk 3]
```diff
@@ -148,3 +148,55 @@ release:ngc-ubi8:
...
+    - 'echo "Signing the image ${IMAGE_NAME}:${IMAGE_TAG}"'
+    - ngc-cli/ngc registry image publish --source ${IMAGE_NAME}:${IMAGE_TAG} ${IMAGE_NAME}:${IMAGE_TAG} --public --discoverable --allow-guest --sign --org nvidia 
+
+sign:ngc:
+  extends:
+    - .sign:ngc
+  parallel:
+    matrix:
+    - SIGN_JOB_NAME: [""]
+      DIST: ["", "ubi8"]
```
Comment (elezar) [inline .nvidia-ci.yml:195]: I would rather stick to `"CI_COMMIT_TAG"` here. There are projects where `""` has a different meaing to `CI_COMMIT_TAG` and I would like the two to remain consistent.

Comment (shivakunv) [reply .nvidia-ci.yml]: changed

**Review by cdesiniotis — APPROVED** [review]

**Comment by cdesiniotis** [comment]
@shivakunv can you squash your commits?

**Comment by shivakunv** [comment]
While correcting GitHub diverged changes, I accidentally closed the PR https://github.com/NVIDIA/k8s-driver-manager/pull/27 . Please review the new PR : https://github.com/NVIDIA/k8s-driver-manager/pull/31

@cdesiniotis @elezar

**Pull request closed without merging**